src/algorithms/synthesis/syrec_line_aware_synthesis.cpp

```diff
@@ -304,14 +303,14 @@ namespace syrec {
         return synthesisOfInversionOk;
     }
 
-    bool LineAwareSynthesis::assignAdd(std::vector<qc::Qubit>& rhs, std::vector<qc::Qubit>& lhs, const AssignStatement::AssignOperation assignOperation) {
+    bool LineAwareSynthesis::assignAdd(std::vector<qc::Qubit>& lhs, std::vector<qc::Qubit>& rhs, const AssignStatement::AssignOperation assignOperation) {
         bool synthesisOfAssignmentOk = true;
         if (const std::optional<BinaryExpression::BinaryOperation> mappedToBinaryOperation = !expOpp.empty() ? tryMapAssignmentToBinaryOperation(assignOperation) : std::nullopt;
             mappedToBinaryOperation.has_value() && *mappedToBinaryOperation == expOpp.top()) {
-            synthesisOfAssignmentOk = increase(annotatableQuantumComputation, rhs, expLhss.top()) && increase(annotatableQuantumComputation, rhs, expRhss.top());
+            synthesisOfAssignmentOk = increase(annotatableQuantumComputation, expLhss.top(), lhs) && increase(annotatableQuantumComputation, expRhss.top(), lhs);
             popExp();
         } else {
-            synthesisOfAssignmentOk = increase(annotatableQuantumComputation, rhs, lhs);
+            synthesisOfAssignmentOk = increase(annotatableQuantumComputation, rhs, lhs); // NOLINT(readability-suspicious-call-argument)
```

Comment:
Maybe this is connected to the previous comment where the arguments are switched (again).